### PR TITLE
Work around Cocoapods dynamic frameworks incompatibility with XCode archive

### LINKED ISSUE #7406: Swift project Build Error when archiving
my project build  is no problem 

but when i try to  Release  app use Xcode archive .  i got this error  .  

``` ruby
Pod::Spec.new do |s|
  s.name             = "YiJianGRPC"
  s.version          = "0.1.6"
  s.summary          = "socket connection with grpc"
  s.ios.deployment_target = '8.0'

  s.platform     = :ios, '8.0'
  s.requires_arc = true

  s.source_files = 'Pod/Classes/*.{h,m}'
  src = "Pod/Classes/"
  dir = "YiJian"

  s.dependency "!ProtoCompiler-gRPCPlugin", "~> 0.14"


  pods_root = '../Pods'


  protoc_dir = "#{pods_root}/!ProtoCompiler"
  protoc = "#{protoc_dir}/protoc"
  plugin = "#{pods_root}/!ProtoCompiler-gRPCPlugin/grpc_objective_c_plugin"



  s.prepare_command = <<-CMD
    mkdir -p #{dir}

    #{protoc} \
       --plugin=protoc-gen-grpc=#{plugin} \
       --objc_out=#{dir} \
       --grpc_out=#{dir} \
       -I #{src} \
       -I #{protoc_dir} \
       #{src}/yijian.proto
  CMD

  s.subspec "Messages" do |ms|
    ms.source_files = "#{dir}/*.pbobjc.{h,m}", "#{dir}/**/*.pbobjc.{h,m}"
    ms.header_mappings_dir = dir
    ms.requires_arc = false
    ms.dependency "Protobuf"
    ms.pod_target_xcconfig = {
     'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1',
   }
  end

  s.subspec "Services" do |ss|
    ss.source_files = "#{dir}/*.pbrpc.{h,m}", "#{dir}/**/*.pbrpc.{h,m}"
    ss.header_mappings_dir = dir
    ss.requires_arc = true
    ss.dependency "gRPC-ProtoRPC"
    ss.dependency "#{s.name}/Messages"
  end


end
```

### PR DESCRIPTION
**Pull request by jcanizales**
Fixes #7406.

“Allow non-modular includes in framework modules” is the magic sentence
that allows pods dependent on RxLibrary to be archived correctly by XCode.

It’s less than ideal, and seems arbitrary (why RxLibrary specifically?),
so we’ll try to produce a minimal case and open an issue with it in the
Cocoapods repo.

### CODE REVIEW

**Comment by jcanizales** [comment]
For some reason Travis isn't showing in the tests box UI - this is the (not yet started) run: https://travis-ci.org/grpc/grpc/builds/145620217

**Comment by jcanizales** [comment]
I've rerun the failed interop tests, although they don't use podspecs so this can't affect them.

**Comment by makdharma** [comment]
test this please

**Comment by jcanizales** [comment]
Wow, Jenkins is supper unhappy today.

**Comment by jcanizales** [comment]
Should we keep rerunning Jenkins? The changed files are podspecs and a readme.

**Comment by jcanizales** [comment]
Ping

**Comment by jcanizales** [comment]
Ping :)

**Comment by muxi** [comment]
Need a bit of explanation before I can understand it :)

**Comment by jcanizales** [comment]
Please give https://github.com/grpc/grpc/issues/7406 a read and ask away :)

**Comment by jcanizales** [comment]
Ping?

File: examples/objective-c/route_guide/RouteGuide.podspec [hunk 1]
```diff
@@ -60,4 +56,11 @@ Pod::Spec.new do |s|
     ss.dependency "gRPC-ProtoRPC"
     ss.dependency "#{s.name}/Messages"
   end
+
+  s.pod_target_xcconfig = {
+    # This is needed by all pods that depend on Protobuf:
+    'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1',
+    # This is needed by all pods that depend on gRPC-RxLibrary:
+    'CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES' => 'YES',
```
Comment (muxi) [inline examples/objective-c/route_guide/RouteGuide.podspec]: As we talked about last week, it is a bit strange if we require any customer who uses our library (like RouteGuide here) to include such a line :)  It LGTM as a temporary fix.

Comment (pgrosu) [inline examples/objective-c/route_guide/RouteGuide.podspec]: Less is more :) We're still in the prerelease time period to streamline it.

**Comment by jcanizales** [comment]
@makdharma for merge-against-Jenkins powers?

**Comment by muxi** [comment]
He's in vacation :)

**Comment by jcanizales** [comment]
@jtattermusch can you merge if Jenkins is unhappy? I've rerun the tests a couple of times already and can't get a green, but this change doesn't affect them.

**Comment by jcanizales** [comment]
Test this please.

**Comment by makdharma** [comment]
test this please

**Comment by jcanizales** [comment]
Let's get this merged, a full month of failing to catch a Jenkins green should be enough :)

**Pull request merged**